### PR TITLE
Refactor glimmer-printer

### PR DESCRIPTION
**Pull request by dcyriller**
#### Description

I'm doing a POC to introduce a `hbsWhitespaceSensitivity` option.

On the way, I noticed that we could group / simplify / extract / remove some code. It might be easier to review it commit by commit. I could split this PR in smaller chunks if you'd like.

#### Checklist

- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-handlebars/printer-glimmer.js [hunk 1]
```diff
@@ -355,18 +248,13 @@ function print(path, options, print) {
...
-            trailingLineBreaksCount,
-            maxLineBreaksToPreserve
-          ),
-        ].filter(Boolean)
-      );
+      return concat([
+        ...generateHardlines(leadingLineBreaksCount, maxLineBreaksToPreserve),
+        n.chars
+          .replace(/^[\s ]+/g, leadingSpace)
+          .replace(/[\s ]+$/, trailingSpace),
```
Comment (fisker) [inline src/language-handlebars/printer-glimmer.js:255]:
```suggestion
          .replace(/^\s+/g, leadingSpace)
          .replace(/\s+$/, trailingSpace),
```

Comment (dcyriller) [reply src/language-handlebars/printer-glimmer.js:255]: strange indeed 🤔

**Comment by thorn0** [comment]
As for `hbsWhitespaceSensitivity`, let's use the existing option `htmlWhitespaceSensitivity`. Otherwise, it'd be strange that there is no `vueWhitespaceSensitivity` and `angularWhitespaceSensitivity`.

**Comment by dcyriller** [comment]
Noted 👍

**Pull request merged**